### ISSUE TITLE
The test `ast-variants` is flaky

### ISSUE THREAD

**Issue by gasche** [ocaml/ocaml#13353]
There is a weird macos-arm64 failure in the CI for #13350 which is clearly unrelated to the PR.

This comes from the `ast-invariants` test
```
2024-08-01T22:56:17.0656510Z > Fatal error: exception Sys_error("/Users/runner/work/ocaml/ocaml/testsuite/tests/asmcomp/_ocamltest/tests/asmcomp/lift_mutable_let_flambda: No such file or directory")
```

The ast-invariants test is a little weird, see [its current definition](https://github.com/ocaml/ocaml/blob/51272e3a89c67fdfd244a30b5c21ac4aaa856808/testsuite/tests/ast-invariants/test.ml): it recursively walks the filesystem, starting from the root of the git repository, and checks all `.ml` files it finds this way.

I believe that the failure comes from a filesystem-level race where the directory walk entered a part of the filesystem that was in the process of being removed after `ocamltest` ran the corresponding tests.

It would be reasonable to restrict the check to only git-versioned .ml files. Another approach would be to blacklist certain directory names that should not be traversed, including `.git` and `_ocamltest`.

**Comment by nojb** [comment]
> It would be reasonable to restrict the check to only git-versioned .ml files.

I run into this problem in the past, and I also thought of this solution. However: I believe that currently it is possible to run the testsuite without an underlying Git repo (ie from a tarball). Is it OK to break this property?

**Comment by gasche** [comment]
No strong opinion. (Maybe the subdirectory blacklist is good enough?)

**Comment by gasche** [comment]
@dra27 proposes to blacklist all `.*` and `_*` subdirectories. Nice!

**Comment by gasche** [comment]
This was fixed by @dra27 in #13455. Thanks!
